From bd57222525bc604c4cf886c17653dbf5d32ac24f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Sat, 2 Nov 2019 15:17:02 +0100
Subject: [PATCH 031/417] firmware: scpi: Don't probe SCP during SCPI driver
 probe

We want to have this driver builtin and be able to load SCP after it
probes at the same time. This allows that.

In the future, we'll load SCP firmware from u-boot, but for now
we want to be able to experiment more freely, by loading SCP after
boot.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/firmware/arm_scpi.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/arm_scpi.c b/drivers/firmware/arm_scpi.c
index 1e1a168ab..5b4c02786 100644
--- a/drivers/firmware/arm_scpi.c
+++ b/drivers/firmware/arm_scpi.c
@@ -813,6 +813,7 @@ struct scpi_ops *get_scpi_ops(void)
 }
 EXPORT_SYMBOL_GPL(get_scpi_ops);
 
+__maybe_unused
 static int scpi_init_versions(struct scpi_drvinfo *info)
 {
 	int ret;
@@ -1011,12 +1012,13 @@ static int scpi_probe(struct platform_device *pdev)
 				scpi_info->cmd_priority);
 	}
 
+	/*
 	ret = scpi_init_versions(scpi_info);
 	if (ret) {
 		dev_err(dev, "incorrect or no SCP firmware found\n");
 		return ret;
 	}
-
+          */
 	if (scpi_info->is_legacy && !scpi_info->protocol_version &&
 	    !scpi_info->firmware_version)
 		dev_info(dev, "SCP Protocol legacy pre-1.0 firmware\n");
-- 
2.35.3

